test/e2e/e2e_test.go

```diff
@@ -222,6 +224,103 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		laraUserName := "laracroft"
+		userNamespace := "laracroft-dev"
+		podName := "test-useraccount-delete-1"
+		laraSignUp := CreateAndApproveSignup(t, hostAwait, laraUserName, memberAwait.ClusterName)
+		VerifyResourcesProvisionedForSignup(t, hostAwait, laraSignUp, "base", memberAwait)
+
+		// when
+		//Create a pod with finalizer in user's namespace, which will block the deletion of namespace.
+		memberPod := corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: userNamespace,
+				Finalizers: []string{
+					"kubernetes",
```

Comment:
Shouldn't we use our own finalizer to make sure we have full control over it and it's not managed by something else?
For example:
```suggestion
					"toolchain-e2e-tests",
```